octopoes/octopoes/repositories/ooi_repository.py

```diff
@@ -503,10 +504,8 @@ def list_neighbours(self, references: Set[Reference], paths: Set[Path], valid_ti
     def save(self, ooi: OOI, valid_time: datetime, end_valid_time: Optional[datetime] = None) -> None:
         # retrieve old ooi
         old_ooi = None
-        try:
+        with contextlib.suppress(ObjectNotFoundException):
             old_ooi = self.get(ooi.reference, valid_time=valid_time)
-        except ObjectNotFoundException:
-            pass
```

Comment:
I think moving  `old_ooi = None` to the exception handler makes the intention of the code clearer than using `contextlib.suppress`. What the code does is set `old_ooi` to the value returned by `self.get` and set it to `None` when `ObjectNotFoundException` is thrown. The same for the save in the other two repository classes.